.github/workflows/test-suite.yml

```diff
@@ -116,6 +116,7 @@ jobs:
         php-version:
           - "8.1"
           - "8.2"
+          - "8.3"
```

Comment:
nvm, it works because composer handles ~8.2 differently, I missed that when I created those constraints. 

> Composer handles tilde constraint differently. Your constraint will translate to >=8.2.0 <9.0.0.

Those constraints should be first changed to `~8.2.0` in order to allow only for `8.2.*` without `8.3`, and then we should explicitly add everywhere ~8.3.0 and only then enable 8.3 on GitHub pipelines.